addons/web/static/src/core/main_components_container.js

```diff
@@ -0,0 +1,37 @@
+/** @odoo-module */
+import { registry } from "./registry";
+import { NotUpdatable, ErrorHandler } from "./utils/components";
+
+const { Component, tags } = owl;
+
+export class MainComponentsContainer extends Component {
```

Comment:
Could this be on its own commit too maybe? Hard to follow the direct link with frontend convertion

Reply:
Sure, I'll move that to its own commit for clarity. Basically I just factored out something that the webclient was already doing so that I could reuse it, but I agree that it's sufficiently self-contained that it can be in its own commit.